src/app/services/conflicts.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class ConflictsService {
+
+  private conflicts: any = {};
+
+  /**
+   * Given the data returned from the API, populate the conflicts cache with
+   * the data for conflicts.
+   * @param {Object} data - The verbatim JSON object returned from the courses API.
+   */
+  public populateConflictsCache(data) {
+    /* NOTE: The data in the conflicts cache MUST be sorted by section ID.
+     * Currently we are assuming the conflicts in the API will always be sorted.
+     * If that is not the case, a sort must be implemented here.
+     */
+    var courselen = data.courses.length;
```

Comment:
Can ES6 / typescriptify this code a bit? At the very least `var`s should be `let`s